manifests/service.pp

```diff
@@ -15,4 +15,11 @@
     }
   }
 
+  $use_servers = join($servers, " ")
+  if $autosync {
+    exec { "/usr/sbin/ntpdate -u -s ${use_servers}":
+     refreshonly => true
```

Comment:
So, is it okay to only trigger any time the config changes? Does it matter if it happens before or after the service restart?

Just because the ntp service is running doesn't mean there is no massive difference... it will just keep that massive difference from creeping :)